src/Branching/perform-migration-from-svn-to-git.md

```diff
@@ -0,0 +1,183 @@
+
+The high level workflow for migrating from SVN to Git is as follows:
+
+* Prepare a migration environment
+* Convert the source SVN repository to a local Git repository
+* (Optional) - Synchronize the local Git repository with any changes from SVN repository while developers continue using SVN
+* Push the local Git repository to a remote Git repository hosted on Visual Studio Team Services
+* Lock SVN repository, synchronize any remaining changes from SVN repository to local Git repository and push final changes to VSTS remote Git repository
+* Developers switch to Git as main source control system
+
+## Prepare a migration environment
+
+A migration environment should be configured on a local workstation. The following software will need to be installed on the migration workstation:
+
+* [Git](https://git-scm.com/downloads)
+* [Subversion](http://subversion.apache.org/packages.html)
+* [git-svn utility](https://www.kernel.org/pub/software/scm/git/docs/git-svn.html) (already part of Git)
+
+You will also need to create a Git repository on your VSTS account to host the converted SVN repository, you may follow [Create a new Git repo in your project
+](https://docs.microsoft.com/en-us/vsts/git/create-new-repo)
+
+## Convert the source SVN repository to a local Git repository
+
+The goal of this step is to convert the source Subversion repository to a local *bare* Git repository. A *bare* Git repository does not have a local working checkout of files that can be modified. This is the recommended format for sharing a Git repository via a remote repository hosted on a service like VSTS.
+
+> [!TIP]
+> *bare* Git repositories are structured differently and given the fact that it doesn't have a working directory prevent direct commit to the repository.
+>![bare git repo](_img/perform-migration-from-svn-to-git/bare-git-repo.png)
+
+### Retrieve a list of all Subversion authors
+
+Subversion just uses the username for each commit. Git stores more data for each author but at a minimal, each commit will need a name and an email. By default, The git-svn tool will list the SVN username in the author and email fields. However, you can create a mapping file for SVN users along with their corresponding Git names and emails.
+
+*Subversion users*
+![Subversion users](_img/perform-migration-from-svn-to-git/svn-log.png)
+
+*Git users*
+![git users](_img/perform-migration-from-svn-to-git/git-log.png)
+
+To extract a list of all SVN users, from the root of your local Subversion checkout, run this command:
+
+```
+svn log -q | awk -F '|' '/^r/ {sub("^ ", "", $2); sub(" $", "", $2); print $2" = "$2" <"$2">"}' | sort -u > authors-transform.txt
```

Comment:
This jumps right into some Unix tools -- and awk, especially, is some crazy magic.  I really wish that our samples would stick to `cmd` and - if that's not possible - PowerShell for the maximum accessibility of the audience.  If we need some crazy scripting, then surely PowerShell can handle this transformation.  And if not, then let's build an app and point people to it.

Reply:
Here is the equivalent command in Powershell. This can be used to make the appropriate replacement in the document. As a side note, we cannot keep it as a CMD as it would still require using Perl or a similar tool.  PS keeps it simple as follows:

svn.exe log --quiet | ? { $_ -notlike '-*' } | % { ($_ -split ' \| ')[1] } | Select-Object -Unique